flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java

```diff
@@ -185,17 +189,14 @@ public void close() {
 
     /** Implementation of a BigQuery write client wrapper. */
     public static class StorageWriteClientImpl implements StorageWriteClient {
-        private static final HeaderProvider USER_AGENT_HEADER_PROVIDER =
-                FixedHeaderProvider.create(
-                        "user-agent", "Apache_Flink_Java/" + FlinkVersion.current().toString());
-
         private final BigQueryWriteClient client;
 
         private StorageWriteClientImpl(CredentialsOptions options) throws IOException {
             BigQueryWriteSettings.Builder settingsBuilder =
                     BigQueryWriteSettings.newBuilder()
                             .setCredentialsProvider(
                                     FixedCredentialsProvider.create(options.getCredentials()))
+                            .setHeaderProvider(USER_AGENT_HEADER_PROVIDER)
                             .setTransportChannelProvider(
                                     BigQueryReadSettings.defaultGrpcTransportProviderBuilder()
                                             .setHeaderProvider(USER_AGENT_HEADER_PROVIDER)
```

Comment:
Why are we setting the user-agent twice in the same client?

Reply:
The existing one is set via the transport channel. This User Agent is not parsed by either of the API's hence we need to se the User Agent via the `setHeaderProvider()`

Reply:
I understand that you are setting the header in a different way now, so my question is do we still need the earlier process of setting it in the transport channel?